### PR TITLE
ci(gitlab): add gitlab-ci support

### PR DESCRIPTION
**Pull request by gperdomor**
Add support for GitLab CI

### Metrics:

The worstcase takes ~38min from start to end
Screen Shot 2019-12-30 at 16 48 23

The worstcase with `--parallel` flag, for some reason takes a longer time, ~50min from start to end
Screen Shot 2019-12-30 at 16 48 11

The distributed CI takes ~15min from start to end
Screen Shot 2019-12-30 at 16 48 35

### CODE REVIEW

**Comment by gperdomor** [comment]
@vsavkin take a look please 🙏

**Comment by gperdomor** [comment]
ping @vsavkin

File: tools/scripts/calculate-commands.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
 const execSync = require('child_process').execSync;
 const isMaster = process.argv[2] === 'False';
-const baseSha = isMaster ? 'origin/master~1' : 'origin/master';
+
+// needed by Gitlab CI
+const remotePrefix = process.env.PREFIX || '';
+
+const baseSha = isMaster
```
Comment (kanafghan) [inline tools/scripts/calculate-commands.js:7]: `baseSha` is always `remotes/origin/master` as `isMaster` is always `false`.

File: tools/scripts/calculate-commands.js [hunk 2]
```diff
@@ -1,10 +1,17 @@
 const execSync = require('child_process').execSync;
 const isMaster = process.argv[2] === 'False';
```
Comment (kanafghan) [inline tools/scripts/calculate-commands.js:2]: `isMaster` is always `false` as `process.argv[2]` will contain the value of `CI_PIPELINE_SOURCE` environment variable and not `False` or `True`.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    <<: *global_cache
+    policy: pull-push
+  before_script:
+    - yarn install --ignore-engines --frozen-lockfile
+  script:
+    - node ./tools/scripts/calculate-commands.js $IS_PR > COMMANDS.txt
+    - echo $(cat COMMANDS.txt)
+  variables:
+    PREFIX: 'remotes/'
+    IS_PR: $CI_PIPELINE_SOURCE == "merge_request_event"
```
Comment (kanafghan) [inline .gitlab-ci.yml:31]: `IS_PR` will become `merge_request_event == "merge_request_event"` and the actual call of the `calculate-commands` script is:

```bash
$ node ./tools/scripts/calculate-commands.js merge_request_event == "merge_request_event"
```

This caused me quite a lot of investigation in order to make the script work with an argument after `$IS_PR`.

**Comment by the-ult** [comment]
Any change this is gonna be merged? And referenced in the nx.dev docs?

**Pull request closed without merging**